src/scripts/dashboardpage.js

```diff
@@ -1,447 +1,928 @@
+                html += '<div class="sessionDeviceName">' + session.DeviceName + "</div>";
+                html += '<div class="sessionAppSecondaryText">' + DashboardPage.getAppSecondaryText(session) + "</div>";
+                html += "</div>";
+                html += "</div>";
+                html += '<div class="sessionNowPlayingTime">' + DashboardPage.getSessionNowPlayingTime(session) + "</div>";
+
+                if (session.TranscodingInfo && session.TranscodingInfo.Framerate) {
+                    html += '<div class="sessionTranscodingFramerate">' + session.TranscodingInfo.Framerate + " fps</div>";
+                } else {
+                    html += '<div class="sessionTranscodingFramerate"></div>';
+                }
+
                 var nowPlayingName = DashboardPage.getNowPlayingName(session);
+
                 if (html += '<div class="sessionNowPlayingInfo" data-imgsrc="' + nowPlayingName.image + '">', html += nowPlayingName.html, html += "</div>", nowPlayingItem && nowPlayingItem.RunTimeTicks) {
-                    html += '<progress class="playbackProgress" min="0" max="100" value="' + 100 * (session.PlayState.PositionTicks || 0) / nowPlayingItem.RunTimeTicks + '"></progress>'
-                } else html += '<progress class="playbackProgress hide" min="0" max="100"></progress>';
-                session.TranscodingInfo && session.TranscodingInfo.CompletionPercentage ? html += '<progress class="transcodingProgress" min="0" max="100" value="' + session.TranscodingInfo.CompletionPercentage.toFixed(1) + '"></progress>' : html += '<progress class="transcodingProgress hide" min="0" max="100"></progress>', html += "</div>", html += "</div>", html += "</div>", html += '<div class="sessionCardFooter cardFooter">', html += '<div class="sessionCardButtons flex align-items-center justify-content-center">';
+                    html += '<progress class="playbackProgress" min="0" max="100" value="' + 100 * (session.PlayState.PositionTicks || 0) / nowPlayingItem.RunTimeTicks + '"></progress>';
+                } else {
+                    html += '<progress class="playbackProgress hide" min="0" max="100"></progress>';
+                }
+
+                if (session.TranscodingInfo && session.TranscodingInfo.CompletionPercentage) {
+                    html += '<progress class="transcodingProgress" min="0" max="100" value="' + session.TranscodingInfo.CompletionPercentage.toFixed(1) + '"></progress>';
+                } else {
+                    html += '<progress class="transcodingProgress hide" min="0" max="100"></progress>';
+                }
+
+                html += "</div>";
+                html += "</div>";
+                html += "</div>";
+                html += '<div class="sessionCardFooter cardFooter">';
+                html += '<div class="sessionCardButtons flex align-items-center justify-content-center">';
                 var btnCssClass;
```

Comment:
shouldn't it be cleaner to do like
```js
var btnCssClass = "hide";
if (session.ServerId && session.NowPlayingItem && session.SupportsRemoteControl && session.DeviceId !== connectionManager.deviceId()) {
    btnCssClass = "";
}
```
?

Reply:
It's used three times though. We should just use proper templating instead of all this html in JS.

Reply:
Agreed.